tests/spark/test_spark_model_export.py

```diff
@@ -639,6 +640,64 @@ def test_model_is_recorded_when_using_direct_save(spark_model_iris):
             assert mlflow.utils.mlflow_tags.MLFLOW_LOGGED_MODELS in current_tags
 
 
+def test_model_logged_via_mlflowdbfs_when_appropriate(spark_model_iris):
+    @dataclass
+    class TestCase:
+        databricks_runtime: bool
+        mlflowdbfs_disabled: bool
+        mlflowdbfs_available: bool
+        expected: str
+
+    testcases = [
+        TestCase(
+            databricks_runtime=True,
+            mlflowdbfs_disabled=False,
+            mlflowdbfs_available=True,
+            expected="mlflowdbfs:///artifacts?run_id={}&path=/model",
+        ),
+        TestCase(
+            databricks_runtime=True,
+            mlflowdbfs_disabled=False,
+            mlflowdbfs_available=False,
+            expected="dbfs://databricks/databricks/mlflow-tracking/a/b/model/sparkml/sparkml",
+        ),
+        TestCase(
+            databricks_runtime=True,
+            mlflowdbfs_disabled=True,
+            mlflowdbfs_available=True,
+            expected="dbfs://databricks/databricks/mlflow-tracking/a/b/model/sparkml/sparkml",
+        ),
+        TestCase(
+            databricks_runtime=False,
+            mlflowdbfs_disabled=False,
+            mlflowdbfs_available=True,
+            expected="dbfs://databricks/databricks/mlflow-tracking/a/b/model/sparkml/sparkml",
+        ),
+    ]
```

Comment:
Awesome! Can we use `pytest.mark.parametrize` to specify the different test cases? That way, we get separate `pytest` invocations of the test cases, which makes it easier to debug failures (in contrast to figuring out which iteration of a loop threw an exception).